src/Medidata.ZipkinTracer.Core/TraceProvider.cs

```diff
@@ -72,11 +73,11 @@ internal TraceProvider(IZipkinConfig config, IOwinContext context = null)
                 headerSampled = context.Request.Headers[SampledHeaderName];
             }
 
-            TraceId = Parse(headerTraceId) ? headerTraceId : GenerateHexEncodedInt64FromNewGuid();
-            SpanId = Parse(headerSpanId) ? headerSpanId : TraceId;
-            ParentSpanId = Parse(headerParentSpanId) ? headerParentSpanId : string.Empty;
+            TraceId = headerTraceId.IsParsableTo128Or64Bit() ? headerTraceId : GenerateHexFromNewGuid(config.Create128BitTraceId);
```

Comment:
@adriancole  Let me just move your comment under this thread.

> my 2p is that unit tests should be where the code exists. punting all the
way to the server to tell if the id is well formatted seems against that
principle.

How would you track the request back to the original request/service call?